### PR TITLE
Export list slow

### PR DESCRIPTION
**Pull request by kaapstorm**
Context: [HI-705](https://dimagi-dev.atlassian.net/browse/HI-705)

The (Django) view was getting called three times. This is because the pagination view model is initialised before list items have been returned. `exportPanelModel` fetches the exports, and the pagination view model also fetches them in order to set `perPage`. When either of the responses arrives, it changes the value of `totalItems` and so the pagination view model sends a third request.

This change allows the subscriptions to only send requests after the export list has been populated by `exportPanelModel`, so that only one request is sent when the page is built. It also preserves current behaviour if the `watchChanges` parameter is not used.

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js [hunk 1]
```diff
@@ -41,12 +41,15 @@ hqDefine('hqwebapp/js/components/pagination', [
             if (params.limitSlug) {
                 self.initialValues.limit = parseInt(self._url.searchParams.get(params.limitSlug));
             }
+            self.watchChanges = (typeof params.watchChanges !== 'undefined') ? params.watchChanges : true;
```
Comment (esoergel) [inline corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js]: This file has parameter docs at the top - could you add a description of `watchChanges` there?

File: corehq/apps/export/templates/export/partials/export_list_controller.html [hunk 2]
```diff
@@ -47,6 +47,7 @@
                                     slug: 'export-list',
                                     perPage: itemsPerPage,
                                     totalItems: totalItems,
+                                    watchChanges: showPagination,
```
Comment (esoergel) [inline corehq/apps/export/templates/export/partials/export_list_controller.html]: What's happening here?  Is `showPagination` `false` in some circumstances?

Comment (kaapstorm) [reply corehq/apps/export/templates/export/partials/export_list_controller.html]: Yes. [export_list.js](https://github.com/dimagi/commcare-hq/blob/53c5c0cb9308069fb50f26750982fbc646e6750f/corehq/apps/export/static/export/js/export_list.js#L235) sets it to `!self.isLoadingPanel() && !self.hasError() && self.exports().length`, so it becomes true when the list has been populated.

Comment (kaapstorm) [reply corehq/apps/export/templates/export/partials/export_list_controller.html]: I'm not delighted with the name "watchChanges". "isReady" might give more of a clue what it's going to do.

**Comment by kaapstorm** [comment]
Pinging @biyeun and @proteusvacuum, who both know more about the pagination view model than I do.

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js [hunk 3]
```diff
@@ -87,6 +83,7 @@ hqDefine('hqwebapp/js/components/pagination', [
                 self.goToPage(Math.max(self.currentPage() - 1, 1), e);
             };
             self.goToPage = function (page, e) {
+                if (!self.isReady()) return;
```
Comment (esoergel) [inline corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js]: :+1: this makes more sense anyways

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js [hunk 4]
```diff
@@ -81,6 +85,7 @@ hqDefine('hqwebapp/js/components/pagination', [
...
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js]: I'm not sure this lives here. Why not put it in the `goToPage` function of the exports code?

**Review by esoergel — APPROVED** [review]
Seems like the initialization here is still a little muddled - I'd expect this behavior to always be the case.  That said, I don't understand this well enough to propose a decent way of changing it.

File: corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js (hunk 4 above)
Comment (kaapstorm) [reply corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js]: That is an excellent idea!

**Comment by kaapstorm** [comment]
Closing in favour of #24677

**Pull request closed without merging**